drivers/soundwire/stream.c

```diff
@@ -281,6 +281,9 @@ static int sdw_program_port_params(struct sdw_master_runtime *m_rt)
 	/* Program transport & port parameters for Slave(s) */
 	list_for_each_entry(s_rt, &m_rt->slave_rt_list, m_rt_node) {
 		list_for_each_entry(p_rt, &s_rt->port_list, port_node) {
+			ret = sdw_slave_set_frequency(s_rt->slave);
```

Comment:
if we want to check the clock frequency, it would be done by writing the same clock scale for all the peripherals on a link, and then we would do it atomically with a bank switch.

In other words, changing the clock is NOT related to the notion of stream. It's something that impacts all peripherals and managers.